share/models/sources.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.db import models
+
+from share.models.ingest import SourceConfig
+from share.models.ingest import NaturalKeyManager
+
+logger = logging.getLogger(__name__)
+__all__ = ('SourceStat', 'SourceStatOAI',)
+
+
+class SourceStat(models.Model):
+    config = models.ForeignKey(SourceConfig, on_delete=models.CASCADE)
+    is_deleted = models.BooleanField(default=False)
+    date_created = models.DateTimeField(auto_now_add=True)
+    response_status_code = models.SmallIntegerField(blank=True, null=True)
+    response_elapsed_time = models.FloatField(blank=True, null=True)
+    response_exception = models.TextField(blank=True, null=True)
+    earliest_datestamp_config = models.DateField(blank=True, null=True)
+    base_url_config = models.TextField()
+    admin_note = models.TextField(blank=True)
+
+    objects = NaturalKeyManager('config.label')
+
+    def natural_key(self):
+        return (self.config.label,)
+
+    def __repr__(self):
+        return '<{}({}, {})>'.format(self.__class__.__name__, self.pk, self.config.label)
+
+    def __str__(self):
+        return '{}: {}'.format(self.config.source.long_title, self.config.label)
+
+
+class SourceStatOAI(SourceStat):
```

Comment:
Just to codify the conversation that @chrisseto and I had. Multi-table inheritance is basically the root of all evil (in most situations) because it forces an implicit join. I'd say if you're going to use these extra fields more often then not I'd put them on the original model and make them nullable. If you're not going to use them very often or they're going to be duplicated often I'd put them in a separate model and FK from the original model to the new model so they can be re-used. If you need them to be independent the FK should point the other way, which is basically the same thing as multi-table inheritance except that it's not going to do the join implicitly.